db/system_keyspace.cc

```diff
@@ -1751,6 +1754,70 @@ future<> set_bootstrap_state(bootstrap_state state) {
     });
 }
 
+class nodetool_status_table : public memtable_filling_virtual_table {
+public:
+    nodetool_status_table() : memtable_filling_virtual_table(build_schema()) {}
+
+    static schema_ptr build_schema() {
+        auto id = generate_legacy_id(NAME, "status");
+        return schema_builder(NAME, "status", std::make_optional(id))
+            .with_column("peer", inet_addr_type, column_kind::partition_key)
+            .with_column("dc", utf8_type)
+            .with_column("up", boolean_type)
+            .with_column("status", utf8_type)
+            .with_column("load", utf8_type)
+            .with_column("tokens", int32_type)
+            .with_column("owns", float_type)
+            .with_column("host_id", uuid_type)
+            .with_version(generate_schema_version(id))
+            .build();
+    }
+
+    future<> execute(memtable& mt) override {
+        auto& ss = service::get_local_storage_service();
+        return ss.get_ownership().then([&] (std::map<gms::inet_address, float> ownership) {
+            const locator::token_metadata& tm = ss.get_token_metadata();
+            gms::gossiper& gs = gms::get_local_gossiper();
+
+            for (auto&& e : gs.endpoint_state_map) {
+                auto endpoint = e.first;
+
+                mutation m(schema(), partition_key::from_single_value(*schema(), data_value(endpoint).serialize_nonnull()));
+                row& cr = m.partition().clustered_row(*schema(), clustering_key::make_empty()).cells();
+
+                set_cell(cr, "up", gs.is_alive(endpoint));
+                set_cell(cr, "status", gs.get_gossip_status(endpoint));
+                set_cell(cr, "load", gs.get_application_state_value(endpoint, gms::application_state::LOAD));
+
+                std::optional<utils::UUID> hostid = tm.get_host_id_if_known(endpoint);
+                if (hostid) {
+                    set_cell(cr, "host_id", hostid);
+                }
+
+                if (tm.get_topology().has_endpoint(endpoint)) {
+                    sstring dc = tm.get_topology().get_location(endpoint).dc;
+                    set_cell(cr, "dc", dc);
+                }
+
+                if (ownership.contains(endpoint)) {
+                    set_cell(cr, "owns", ownership[endpoint]);
+                }
+
+                set_cell(cr, "tokens", int32_t(tm.get_tokens(endpoint).size()));
+
+                mt.apply(m);
+            }
+        });
+    }
+};
+
+static thread_local std::vector<std::unique_ptr<virtual_table>> virtual_tables;
+
+void register_virtual_tables() {
```

Comment:
we should probably pass database as a param to register_virtual_tables(). soon it will be required anyway to register commands which read from database object and respective tables.

Reply:
Shouldn't we wait until it's indeed required?

Reply:
We can do it later.